### PR TITLE
LogReader missing files counter

### PR DESCRIPTION
**Pull request by StefanLekanic-SYRMIA**
#32797

### CODE REVIEW

**Comment by sshane** [comment]
can you implement it in the error message? You can also avoid adding too many lines by incorporating a check to ensure that missing files == 0 or similar

**Comment by StefanLekanic-SYRMIA** [comment]
I hope that i understood properly what were you looking for...

**Comment by StefanLekanic-SYRMIA** [comment]
By checking the code, i figured out that there is some possible redundancy, in way that there is needlessly checking of qlogs. It probably could be optimized.

File: tools/lib/logreader.py [hunk 1]
```diff
@@ -92,18 +92,32 @@ def default_valid_file(fn: LogPath) -> bool:
...
-            for (rlog, qlog) in zip(rlog_paths, qlog_paths, strict=True)]
+  # Count missing files
+  missing_files_count = count_missing_files(rlog_paths, qlog_paths, valid_file)
+
+  if missing_files_count != 0:
+    cloudlog.error(f"Error: {missing_files_count} segments are missing either rlogs or qlogs.")
+    # auto select logs based on availability
+    if any(rlog is None or not valid_file(rlog) for rlog in rlog_paths) and all(qlog is not None and valid_file(qlog) for qlog in qlog_paths):
+      if interactive:
+        if input("Some rlogs were not found, would you like to fallback to qlogs for those segments? (y/n) ").lower() != "y":
```
Comment (sshane) [inline tools/lib/logreader.py:103]: just integrate the count into these error messages

File: tools/lib/logreader.py [hunk 2]
```diff
@@ -92,18 +92,32 @@ def default_valid_file(fn: LogPath) -> bool:
...
-
-    return [rlog if valid_file(rlog) else (qlog if valid_file(qlog) else None)
-            for (rlog, qlog) in zip(rlog_paths, qlog_paths, strict=True)]
+  # Count missing files
+  missing_files_count = count_missing_files(rlog_paths, qlog_paths, valid_file)
+
+  if missing_files_count != 0:
+    cloudlog.error(f"Error: {missing_files_count} segments are missing either rlogs or qlogs.")
+    # auto select logs based on availability
+    if any(rlog is None or not valid_file(rlog) for rlog in rlog_paths) and all(qlog is not None and valid_file(qlog) for qlog in qlog_paths):
```
Comment (sshane) [inline tools/lib/logreader.py:101]: can we remove this line if we check `missing_files_count` right above?

**Comment by StefanLekanic-SYRMIA** [comment]
Only way to do that things is to count rlogs and qlogs separately. With that we can use more intuitive if statement, and incorporate right values for missing logs.

File: tools/lib/logreader.py (hunk 1 above)
Comment (StefanLekanic-SYRMIA) [reply tools/lib/logreader.py]: We count all logs, and it is not possible to say does missing just rlogs, like that...

File: tools/lib/logreader.py (hunk 2 above)
Comment (StefanLekanic-SYRMIA) [reply tools/lib/logreader.py]: No, because there could be scenario with qlogs missing.

**Comment by sshane** [comment]
If it's possible, this should be a couple line change. We don't want to add this many lines for such a feature

**Comment by StefanLekanic-SYRMIA** [comment]
Then we can just count missing rlogs, cause you expecting all qlogs to be there, and does not returning them anywhere.

**Pull request closed without merging**